gvr-simplephysics/app/src/main/java/org/gearvrf/gvrsimlephysics/main/MainHelper.java

```diff
@@ -0,0 +1,188 @@
+public class MainHelper {
+    private static final float CYLINDER_MASS = 0.3f;
+    private static final float BALL_MASS = 0.5f;
+    private static final int COLLISION_GROUP_INFINITY_GROUND = 0;
+    private static final int COLLISION_GROUP_GROUND = 1;
+    private static final int COLLISION_GROUP_BALL = 3;
+    public static final int COLLISION_GROUP_CYLINDER = 2;
+
+    public static GVRCollisionMatrix collisionMatrix;
+
+    static {
+        collisionMatrix = new GVRCollisionMatrix();
+
+        collisionMatrix.setCollisionFilterMask(COLLISION_GROUP_INFINITY_GROUND, (short) 0x0);
+
+        collisionMatrix.enableCollision(COLLISION_GROUP_CYLINDER, COLLISION_GROUP_GROUND);
+        collisionMatrix.enableCollision(COLLISION_GROUP_BALL, COLLISION_GROUP_GROUND);
+        collisionMatrix.enableCollision(COLLISION_GROUP_BALL, COLLISION_GROUP_CYLINDER);
+    }
+
+    public static GVRSceneObject createPointLight(GVRContext context, float x, float y, float z) {
+        GVRSceneObject lightObject = new GVRSceneObject(context);
+        GVRPointLight light = new GVRPointLight(context);
+
+       float ambientIntensity = 0.5f;
+       float diffuseIntensity = 1.0f;
+
+        light.setAmbientIntensity(1.0f * ambientIntensity, 0.95f * ambientIntensity, 0.83f * ambientIntensity, 0.0f);
+        light.setDiffuseIntensity(1.0f * diffuseIntensity, 0.95f * diffuseIntensity, 0.83f * diffuseIntensity, 0.0f);
+        light.setSpecularIntensity(0.0f, 0.0f, 0.0f, 0.0f);
+
+        //lightObject.getTransform().setScale(1.0f, 1.0f, 5.0f);
+        lightObject.getTransform().setPosition(x, y, z);
+        lightObject.attachLight(light);
+        return lightObject;
+    }
+
+   public static GVRSceneObject createDirectLight(GVRContext context, float x, float y, float z) {
+        Future<GVRTexture> texture = context.loadFutureTexture(new GVRAndroidResource(context, R.drawable.yellow));
+        GVRSceneObject lightObject = new GVRSceneObject(context);//GVRSphereSceneObject(context, true, texture);
+        GVRDirectLight light = new GVRDirectLight(context);
+
+        light.setCastShadow(true);
+
+       float ambientIntensity = 0.1f;
+       float diffuseIntensity = 1.0f;
+
+        light.setAmbientIntensity(1.0f * ambientIntensity, 0.95f * ambientIntensity, 0.83f * ambientIntensity, 0.0f);
+        light.setDiffuseIntensity(1.0f * diffuseIntensity, 0.95f * diffuseIntensity, 0.83f * diffuseIntensity, 0.0f);
+        light.setSpecularIntensity(0.0f, 0.0f, 0.0f, 0.0f);
+
+        lightObject.getTransform().setPosition(x, y, z);
+        lightObject.attachLight(light);
+        return lightObject;
+    }
+
+    public static GVRSceneObject createGround(GVRContext context, float x, float y, float z) {
+        Future<GVRTexture> texture = context.loadFutureTexture(new GVRAndroidResource(context, R.drawable.orange));
+        GVRMaterial material = new GVRMaterial(context);
+
+        GVRSceneObject groundObject = new GVRCubeSceneObject(context, true, texture);
```

Comment:
You can replace 88 through 99 with the following to avoid the setMainTexture line that masks a bug in the framework:

```
        GVRSceneObject groundObject = new GVRCubeSceneObject(context, true, texture);

        groundObject.getRenderData().getMaterial().setTexture("diffuseTexture", texture);
        groundObject.getTransform().setScale(15.0f, 0.5f, 15.0f);
        groundObject.getTransform().setPosition(x, y, z);

        //set phong Shader
        groundObject.getRenderData().setShaderTemplate(GVRPhongShader.class);
```

There is a bug if you create a material and assign it to the render data. Have observed this issue multiple times now. Will file a ticket.